tests/unittests/benchmark/task/profet/test_profet_task.py

```diff
@@ -0,0 +1,375 @@
+from pathlib import Path
+from typing import ClassVar, Type
+import pytest
+
+from orion.benchmark.task.base import BaseTask
+from orion.benchmark.task.profet.profet_task import (
+    MetaModelTrainingConfig,
+    ProfetTask,
+    download_data,
+    load_data,
+)
+from orion.algo.space import _Discrete
+from typing import Dict, Tuple
+import numpy as np
+import torch
+
+from logging import getLogger as get_logger
+from .conftest import REAL_PROFET_DATA_DIR, requires_profet_data
+
+logger = get_logger(__name__)
+shapes: Dict[str, Tuple[Tuple[int, ...], Tuple[int, ...], Tuple[int, ...]]] = {
+    "fcnet": ((600, 6), (27, 600), (27, 600)),
+    "forrester": ((10, 1), (9, 10), (9, 10)),
+    "svm": ((200, 2), (26, 200), (26, 200)),
+    "xgboost": ((800, 8), (11, 800), (11, 800)),
+}
+y_min: Dict[str, float] = {
+    "fcnet": 0.0,
+    "forrester": -18.049155413936802,
+    "svm": 0.0,
+    "xgboost": 0.0,
+}
+y_max: Dict[str, float] = {
+    "fcnet": 1.0,
+    "forrester": 14718.31848526001,
+    "svm": 1.0,
+    "xgboost": 3991387.335843141,
+}
+c_min: Dict[str, float] = {
+    "fcnet": 0.0,
+    "forrester": -18.049155413936802,
+    "svm": 0.0,
+    "xgboost": 0.0,
+}
+c_max: Dict[str, float] = {
+    "fcnet": 14718.31848526001,
+    "forrester": 14718.31848526001,
+    "svm": 697154.4010462761,
+    "xgboost": 5485.541382551193,
+}
+
+
+@pytest.fixture(autouse=True)
+def load_fake_data(monkeypatch, tmp_path_factory):
+    """ Fixture that prevents attempts to download the true Profet datasets, and instead generates
+    random data with the same shape.
+    """
+
+    import orion.benchmark.task.profet.profet_task
+
+    real_load_data = orion.benchmark.task.profet.profet_task.load_data
+
+    def _load_data(path: Path, benchmark: str):
+        if path == REAL_PROFET_DATA_DIR and REAL_PROFET_DATA_DIR.exists():
+            # Return real datasets.
+            logger.info(f"Testing using the real Profet datasets.")
+            return real_load_data(path, benchmark=benchmark)
+        # Generate fake datasets.
+        logger.info(f"Warning: Using random data instead of the actual profet training data.")
+        x_shape, y_shape, c_shape = shapes[benchmark]
+        X = np.random.rand(*x_shape)
+        min_y = y_min[benchmark]
+        max_y = y_max[benchmark]
+        Y = np.random.rand(*y_shape) * (max_y - min_y) + min_y
+        min_c = c_min[benchmark]
+        max_c = c_max[benchmark]
+        C = np.random.rand(*c_shape) * (max_c - min_c) + min_c
+        return X, Y, C
+
+    monkeypatch.setattr(orion.benchmark.task.profet.profet_task, "load_data", _load_data)
+    # NOTE: Need to set the item in the globals because it might already have been imported.
+    monkeypatch.setitem(globals(), "load_data", _load_data)
+
+
+@pytest.mark.timeout(10)
+@requires_profet_data
+@pytest.mark.parametrize("benchmark", ["fcnet", "forrester", "svm", "xgboost"])
+def test_download_fake_datasets(tmp_path_factory, benchmark: str, load_fake_data):
```

Comment:
So your goal is to test the fixture or load_data?

Reply:
Test the fixture

Reply:
Ok. It takes 10s to run due to the timeout? Is that enough to test it properly?

Reply:
Sorry, I don't quite understand your question!
But idk if that answers it: loading the real data can still take a little bit of time, (<10 secs), so I set the timeout to 15 seconds here.

Reply:
I think it does answer my question. Just to make sure I understand correctly, you are saying that with a timeout of 15s the downloads are completing in time without timeouts? That's good.

Reply:
(hopefully) Fixed: The tests for this now have a better docstring, and the options are made a bit more explicit. Let me know if you have any more questions.

Reply:
Is this now defined with `mock_load_data`?

Reply:
Oh, you broke it into 2 tests? I can't find this test anymore.

Reply:
Yeah, got broken down into:
- `test_mock_load_data_fixture_when_data_isnt_available`
- `test_mock_load_data_fixture_when_real_data_available`

Reply:
Yep, that's good!